support/cas-server-support-consent-couchdb/src/main/java/org/apereo/cas/couchdb/CouchDbConsentDecision.java

```diff
@@ -0,0 +1,90 @@
+package org.apereo.cas.couchdb;
+
+import org.apereo.cas.consent.ConsentDecision;
+import org.apereo.cas.consent.ConsentReminderOptions;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+
+import java.security.SecureRandom;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * This is {@link CouchDbConsentDecision}.
+ *
+ * @author Timur Duehr
+ * @since 6.0.0
+ */
+@Getter
+@Setter
+public class CouchDbConsentDecision extends ConsentDecision {
+    @JsonProperty("_rev")
+    private String rev;
+
+    @JsonProperty("_id")
+    private String cid;
+
+    @JsonCreator
+    public CouchDbConsentDecision(@JsonProperty("_id") final String cid,
+                                        @JsonProperty("_rev") final String rev,
+                                        @JsonProperty("id") final long id,
+                                        @JsonProperty("principal") final String principal,
+                                        @JsonProperty("service") final String service,
+                                        @JsonProperty("createdDate") final LocalDateTime createdDate,
+                                        @JsonProperty("options") final ConsentReminderOptions options,
+                                        @JsonProperty("reminder") final long reminder,
+                                        @JsonProperty("reminderTimeUnit") final ChronoUnit reminderTimeUnit,
+                                        @JsonProperty("attributes") final String attributes) {
+        this.cid = cid;
+        this.rev = rev;
+        setId(id);
+        setPrincipal(principal);
+        setService(service);
+        setCreatedDate(createdDate);
+        setOptions(options);
+        setReminder(reminder);
+        setReminderTimeUnit(reminderTimeUnit);
+        setAttributes(attributes);
+    }
+
+    /**
+     * Copy constructor.
+     * @param c Consent decision to copy from.
+     */
+    @SneakyThrows
+    public CouchDbConsentDecision(final ConsentDecision c) {
+        setAttributes(c.getAttributes());
+        setPrincipal(c.getPrincipal());
+        setCreatedDate(c.getCreatedDate());
+        setId(c.getId());
+        setOptions(c.getOptions());
+        setReminder(c.getReminder());
+        setReminderTimeUnit(c.getReminderTimeUnit());
+        setService(c.getService());
+        if (getId() < 0) {
+            setId(SecureRandom.getInstanceStrong().nextLong());
```

Comment:
nextLong could return a negative, right?

Reply:
You may want to avoid using SecureRandom natively, and use the CAS wrapper class instead.

Reply:
Done.